RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplateReco.cc

```diff
@@ -175,19 +174,26 @@ int SiPixelTemplateReco::PixelTempReco1D(int id,
 
   // The minimum chi2 for a valid one pixel cluster = pseudopixel contribution only
 
-  const double mean1pix = {0.100}, chi21min = {0.160};
+  const double mean1pix = {0.100};
+#ifdef SI_PIXEL_TEMPLATE_STANDALONE
+  const double chi21min = {0.};
+#else
+  const double chi21min = {0.160};
+#endif
 
   // First, interpolate the template needed to analyze this cluster
   // check to see of the track direction is in the physical range of the loaded template
 
-  if (!templ.interpolate(id, cotalpha, cotbeta, locBz, locBx)) {
-    if (theVerboseLevel > 2) {
-      LOGDEBUG("SiPixelTemplateReco") << "input cluster direction cot(alpha) = " << cotalpha
-                                      << ", cot(beta) = " << cotbeta << ", local B_z = " << locBz
-                                      << ", local B_x = " << locBx << ", template ID = " << id
-                                      << ", no reconstruction performed" << ENDL;
+  if (id > 0) {  //if id == 0 bypass interpolation (used in calibration)
```

Comment:
Can this be originating the differences in the FastSim? (just guessing....)

Reply:
I don't think fastsim uses the template reco. I think the differences stem from the calls made [here](https://github.com/cms-sw/cmssw/blob/master/FastSimulation/TrackingRecHitProducer/src/PixelTemplateSmearerBase.cc). I'm trying to understand now

Reply:
Possibly.

I want to comment on FastSim stuff: we are actually responsible for the FastSim code.  We looked at the diffs, and they are very small.

That being said, the improved template calibration infrastructure is actually going to help FastSim since we can merge the code that makes the FastSim pixel rechit resolution files.  This is vital especially for HL-LHC, since the current FastSim resolution files there have the wrong electronics response (which is included in this PR...)

Reply:
FastSim does *not* use the template reco, but it is querying the template object to figure things out that are otherwise not simulated (e.g. the typical dimensions of the cluster and such).